### ISSUE TITLE
Scanning/hovering over profiles with cyrillic text causes disconnection in the client

### ISSUE THREAD

**Issue by santernn** [Total-RP/Total-RP-3#886]
## Describe the issue
Scanning/hovering over profiles with cyrillic text causes the session to disconnect. This seems to mostly occur while doing instanced content (Raids, BGs, Arena, M+ etc.)

## Expected behaviour
The addon should be able to function as normal even with Russian players being present in groups.

## Issue checklist
Please select any applicable options from the below list.

> You can replace the "[ ]" with "[x]" for each option below to tick it.

- [x ] The issue occurs with the [latest version] of the addon.
- [x ] The issue occurs with Total RP 3 as the only enabled addon.
- [ ] The issue occurs only with other addon(s) enabled (please provide a list).

## Steps to reproduce
Give us a list of steps to reproduce the issue with.

1. Have TRP3 enabled
2. Enter any instanced content with a group that has Russian players
3. Select or mouse over a player that has Cyrillic text in their character, guild name or TRP profile
4. Client disconnects shortly after
5. Issue ceases to reoccur once TRP3 is disabled and the UI is reloaded

## Additional files
Will provide logs, was missing bugsack at the time this was happening
Issue was reported and reposted by other players in the following threads:
https://eu.forums.blizzard.com/en/wow/t/trp3-mrp-causing-pvp-disconnects/512824/5
https://eu.forums.blizzard.com/en/wow/t/repeated-disconnects-in-solo-shuffle/512237/5

---

[error logs]: https://www.curseforge.com/wow/addons/bugsack
[profile data files]: https://github.com/Total-RP/Total-RP-3/wiki/Saved-Variables
[latest version]: https://www.curseforge.com/wow/addons/total-rp-3/files

**Comment by Meorawr** [comment]
It's an issue Blizzard are looking into. Not one that we can really fix on our end short of blacklisting an entire region worth of players from the addon.

For clarification - is it a _total client crash_ (as in, straight to desktop) or just a disconnect? I've only seen the latter.

**Comment by santernn** [comment]
@Meorawr good point, it was disconnecting, not the client crashing. I edited accordingly
Any clue based on previous cases how long does Blizz take to fix such issues?

**Comment by Meorawr** [comment]
I'll see if I can coax an update out of someone later today. I suspect it's relatively high priority, but the server team were only made aware of the issue quite late yesterday so probably haven't had much chance to go trawling through server logs to figure out the cause.

**Comment by santernn** [comment]
Thanks!

**Comment by Meorawr** [comment]
We've stuck a temporary workaround into the just-released version 2.8.3 that should mitigate the disconnects until Blizzard resolve it. We didn't really want to be in a position where the deployment of their fix gets pushed into next week owing to the launch if timerunning and the general proximity to the weekend.

As a consequence of our workaround, transfers won't work cross-realm when requesting profiles from players whose combined character and realm names exceed a certain length. That workaround will disable itself in two weeks, which one would hope is plenty of time for the issue to be properly resolved by Blizzard. Also, you may still disconnect if _other_ installed addons try to communicate with problematic players.

**Comment by Meorawr** [comment]
To also update here - I'd strongly recommend updating to 2.8.4 where we've made the "temporary" workaround more permanent. 

When we'd made the temporary workaround we had expected the issue given its severity would have been resolved in either of the two previous weekly maintenance restarts, however the issue remains as yet unresolved on live servers.

When we released 2.8.3 we'd set it up the workaround such that it would disable itself on June 1st. The rationale behind this was mostly to deal with the problem where users may be a bit slow when it comes to updating addons, and us not wanting users on the 2.8.3 version to be left in a state where profile transfers to certain players were blocked for arbitrary and unclear reasons.

However, as the core issue remains unfixed on live servers and there's not really any guarantee on when that'll change we've more or less been forced to remove that timer completely. The net result is that 2.8.3 as of today may once again run into disconnects, but 2.8.4 onwards never will.
